src/common/proxy_wasm/ngx_proxy_wasm_properties.c

```diff
@@ -878,23 +880,28 @@ ngx_proxy_wasm_properties_get_host(ngx_proxy_wasm_ctx_t *pwctx,
     hpn = (host_props_node_t *)
               ngx_str_rbtree_lookup(&pwctx->host_props_tree, path, hash);
 
-    if (hpn) {
+    if (hpn && (hpn->is_const || pwctx->host_prop_getter == NULL)) {
         value->data = hpn->value.data;
         value->len = hpn->value.len;
 
+        if (hpn->negative_cache) {
+            *negative_cache = 1;
```

Comment:
Can do! Generally I'd avoid inventing such purpose-specific meanings to API-provided constants, but looks like it's a common pattern in Nginx code, and it will streamline it anyway in a way which looks clear enough. :+1: